wdl-analysis/src/handlers/completions.rs

```diff
@@ -0,0 +1,453 @@
+        if token.kind() == SyntaxKind::Ident {
+            if let Some(ns) = document.namespace(token.text()) {
+                for task in ns.document().tasks() {
+                    items.push(CompletionItem {
+                        label: task.name().to_string(),
+                        kind: Some(CompletionItemKind::FUNCTION),
+                        detail: Some(format!("task {}", task.name())),
+                        ..Default::default()
+                    })
+                }
+                if let Some(workflow) = ns.document().workflow() {
+                    items.push(CompletionItem {
+                        label: workflow.name().to_string(),
+                        kind: Some(CompletionItemKind::FUNCTION),
+                        detail: Some(format!("workflow {}", workflow.name())),
+                        ..Default::default()
+                    });
+                }
+
+                return Ok(());
+            }
+        }
+    }
+
+    // NOTE: We do type evaluation only for non namespaces or complex types
+
+    let Some(target_node) = target_element.as_node() else {
+        return Ok(());
+    };
+
+    let Some(target_expr) = Expr::cast(target_node.clone()) else {
+        return Ok(());
+    };
+
+    let Some(scope) = document.find_scope_by_position(node.span().start()) else {
+        bail!("could not find scope for access expression")
+    };
+
+    let mut ctx = TypeEvalContext { scope, document };
+    let mut evaluator = ExprTypeEvaluator::new(&mut ctx);
+    let target_type = evaluator.evaluate_expr(&target_expr).unwrap_or(Type::Union);
+
+    match target_type {
+        Type::Compound(CompoundType::Struct(s), _) => {
+            for (name, ty) in s.members() {
+                items.push(CompletionItem {
+                    label: name.to_string(),
+                    kind: Some(CompletionItemKind::FIELD),
+                    detail: Some(ty.to_string()),
+                    ..Default::default()
+                });
+            }
+        }
+        Type::Call(call) => {
+            for (name, output) in call.outputs() {
+                items.push(CompletionItem {
+                    label: name.to_string(),
+                    kind: Some(CompletionItemKind::FIELD),
+                    detail: Some(output.ty().to_string()),
+                    ..Default::default()
+                });
+            }
+        }
+        Type::Compound(CompoundType::Pair(p), _) => {
+            items.push(CompletionItem {
+                label: "left".to_string(),
+                kind: Some(CompletionItemKind::FIELD),
+                detail: Some(p.left_type().to_string()),
+                ..Default::default()
+            });
+
+            items.push(CompletionItem {
+                label: "right".to_string(),
+                kind: Some(CompletionItemKind::FIELD),
+                detail: Some(p.right_type().to_string()),
+                ..Default::default()
+            });
```

Comment:
_This was done as an experiment_

TODO: I don't know how but if time permits in a future PR i would want to do _inferred_ completions for maps.

```wdl
version 1.2

workflow test_map {
  Map[String, Int] string_to_int = { "a": 1, "b": 2 }
  Map[File, Array[Int]] file_to_ints = {
    "/path/to/file1": [0, 1, 2],
    "/path/to/file2": [9, 8, 7]
  }

  output {
    Int b = string_to_int["b"]  # evaluates to 2
    Array[Int] ints = file_to_ints["/path/to/file1"]  # evaluates to [0, 1, 2]
  }
}
```

so someone typing `file_to_ints["`, LSP should suggest  `"/path/to/file1"`, `"/path/to/file2"` in completions list. Obvisouly this would only work if they can be inferred.

Not a scope for this PR tho